assignments/arrays.js

```diff
@@ -66,31 +66,71 @@ let inventory = [{"id":1,"car_make":"Lincoln","car_model":"Navigator","car_year"
 console.log(`Car 33 is a *car year goes here* *car make goes here* *car model goes here*` );
 
 
+for (let i = 0; i < inventory.length; i++) {
+	console.log(`Car ${inventory[i]["id"]} is a ${inventory[i]["car_year"]} ${inventory[i]["car_make"]} ${inventory[i]["car_model"]}`);
+}
 
 // ==== Challenge 2 ====
 // The dealer needs the information on the last car in their inventory.  What is the make and model of the last car in the inventory?  Log the make and model into the console.
 let lastCar = 0;
 console.log();
 
+let lastCar = 0;
+lastCar = inventory[inventory.length - 1]["car_make"] + " " + inventory[inventory.length - 1]["car_model"];
+console.log(lastCar);
+
 // ==== Challenge 3 ====
 // The marketing team wants the car models listed alphabetically on the website. Sort all the car model names into alphabetical order and log the results in the console
 let carModels = [];
 console.log();
 
+let carModels = [];
+for (let i = 0; i < inventory.length; i++) {
+	carModels.push(inventory[i]["car_model"]);
+}
+carModels.sort();
+for (let i = 0; i < carModels.length; i++) {
+	console.log(carModels[i]);	
+}
+
 // ==== Challenge 4 ====
 // The accounting team needs all the years from every car on the lot. Create a new array from the dealer data containing only the car years and log the result in the console.
 let carYears = [];
 console.log();
 
+let carYears = [];
+for (let i = 0; i < inventory.length; i++) {
+	carYears.push(inventory[i]["car_year"]);
+}
+for (let i = 0; i < carYears.length; i++) {
+	console.log(carYears[i]);	
+}
+
+
 // ==== Challenge 5 ====
 // The car lot manager needs to find out how many cars are older than the year 2000. Using the carYears array you just created, find out how many cars were made before the year 2000 by populating the array oldCars and logging it's length.
 let oldCars =[];
 console.log(); 
 
+let oldCars =[];
+for (let i = 0; i < carYears.length; i++) {
+	if (carYears[i] < 2000) oldCars.push(carYears[i]);
+}
+console.log(oldCars.length); 
+
 // ==== Challenge 6 ====
 // A buyer is interested in seeing only BMW and Audi cars within the inventory.  Return an array that only contains BMW and Audi cars.  Once you have populated the BMWAndAudi array, use JSON.stringify() to show the results of the array in the console.
 let BMWAndAudi =[];
 console.log();
 
+for (let i = 0; i < inventory.length; i ++) {
+	if (inventory[i]["car_make"] === "Audi" || inventory[i]["car_make"] === "BMW") BMWAndAudi.push(inventory[i]);
```

Comment:
You definitely want the BMWAndAudi.push(inventory[i]); in brackets, because of block scopes.

https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Statements/block